### PR TITLE
[libc++] <experimental/simd> Add implicit type conversion constructor for class simd/simd_mask

### PR DESCRIPTION
**Pull request by joy2myself**

### CODE REVIEW

**Comment by joy2myself** [comment]
Gentle ping. @philnik777

File: libcxx/include/experimental/__simd/simd.h [hunk 1]
```diff
@@ -46,6 +46,16 @@ class simd {
   template <class _Up, enable_if_t<__can_broadcast_v<value_type, __remove_cvref_t<_Up>>, int> = 0>
   _LIBCPP_HIDE_FROM_ABI simd(_Up&& __v) noexcept : __s_(_Impl::__broadcast(static_cast<value_type>(__v))) {}
 
+  // implicit type conversion constructor
+  template <class _Up,
+            class = enable_if_t<!is_same_v<_Up, _Tp> && is_same_v<abi_type, simd_abi::fixed_size<size()>> &&
+                                __is_non_narrowing_convertible_v<_Up, value_type>>>
```
Comment (philnik777) [inline libcxx/include/experimental/__simd/simd.h:52]:
```suggestion
            enable_if_t<!is_same_v<_Up, _Tp> && is_same_v<abi_type, simd_abi::fixed_size<size()>> &&
                        __is_non_narrowing_convertible_v<_Up, value_type>, int> = 0>
```

File: libcxx/test/std/experimental/simd/simd.class/simd_ctor_conversion.pass.cpp [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    std::array<T, array_size> expected_value;
+    for (size_t i = 0; i < array_size; ++i)
+      expected_value[i] = static_cast<T>(i);
+
+    types::for_each(arithmetic_no_bool_types(), ConversionHelper<T, SimdAbi, array_size>(expected_value));
+  }
+};
+
+template <class U, class T, class SimdAbi = ex::simd_abi::compatible<T>, class = void>
+struct has_conversion_ctor : std::false_type {};
```
Comment (philnik777) [inline libcxx/test/std/experimental/simd/simd.class/simd_ctor_conversion.pass.cpp:53]: `is_convertible`?

**Review by philnik777 — APPROVED** [review]
LGTM % comments.

File: libcxx/test/std/experimental/simd/simd.class/simd_ctor_conversion.pass.cpp (hunk 2 above)
Comment (joy2myself) [reply libcxx/test/std/experimental/simd/simd.class/simd_ctor_conversion.pass.cpp:53]: I tried it but it seems `is_convertible` can not accurately test the type conversion constructor. When `U` is the same as `T`, `simd[_mask]<U, Abi>` is the same as `simd[_mask]<T, Abi>`. And the default copy constructor make them convertible no matter what the `Abi` is.

Comment (joy2myself) [reply libcxx/test/std/experimental/simd/simd.class/simd_ctor_conversion.pass.cpp]: Sorry, my previous understanding may not right. The new commit may be what you want.

**Pull request merged**